### PR TITLE
[#11] Use optparse-applicative for parsing configuration

### LINKED ISSUE #11: Add logging CLI flag via `optparse-applicative`
So later we can enable or disable logging in our application via this flag.

### PR DESCRIPTION
**Pull request by cronokirby**
resolves #11 
This introduces a `Context` type to contain the settings for the program, and some basic
scaffolding to parse and consume this record.

Perhaps we should move this to another module, like `Hintman.Context` to easily share it across other things?

### CODE REVIEW

File: src/Hintman.hs [hunk 1]
```diff
@@ -2,5 +2,29 @@ module Hintman
        ( someFunc
        ) where
 
+import Options.Applicative
```
Comment (vrom911) [inline src/Hintman.hs]: Can you please explicitly write imports from this library?

File: src/Hintman.hs [hunk 2]
```diff
@@ -2,5 +2,29 @@ module Hintman
...
+
+-- | Represents the application settings
+data Context = Context
+    { contextLogging :: Bool -- ^ If True, enable logging
+    }
+
+parseContext :: Parser Context
+parseContext = Context
+    <$> switch
+        (  long "logging"
```
Comment (vrom911) [inline src/Hintman.hs]: We can add `short 'l'` as well, how do you think?

File: src/Hintman.hs [hunk 3]
```diff
@@ -2,5 +2,29 @@ module Hintman
...
+import Options.Applicative
+
+-- | Represents the application settings
+data Context = Context
+    { contextLogging :: Bool -- ^ If True, enable logging
+    }
+
+parseContext :: Parser Context
+parseContext = Context
+    <$> switch
```
Comment (vrom911) [inline src/Hintman.hs]: Probably we will have some other cli options in the future, so I guess it makes sense to move logging parser to the separate function (`loggingP` or any other name)

File: src/Hintman.hs (hunk 1 above)
Comment (cronokirby) [reply src/Hintman.hs]: sure, there's quite a few of them though :P

File: src/Hintman.hs [hunk 4]
```diff
@@ -2,5 +2,14 @@ module Hintman
        ( someFunc
        ) where
 
+import Hintman.Cli (Context(..), cliContext)
+
 someFunc :: IO ()
-someFunc = putStrLn ("someFunc" :: String)
+someFunc = cliContext >>= runOn
```
Comment (ChShersh) [inline src/Hintman.hs]: I feel like `Hintman` module should just reexport single `runHintman` function that parse CLI and runs server. Could you do this in this PR as well?

File: hintman.cabal [hunk 5]
```diff
@@ -24,10 +24,12 @@ library
   hs-source-dirs:      src
   exposed-modules:     Hintman
                            Hintman.App
```
Comment (vrom911) [inline hintman.cabal]:
```suggestion
```

**Pull request merged**